streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSegmentedBytesStore.java

```diff
@@ -234,7 +275,11 @@ public void put(final Bytes key,
 
     @Override
     public byte[] get(final Bytes key) {
-        final S segment = segments.getSegmentForTimestamp(keySchema.segmentTimestamp(key));
+        final long timestampFromKey = keySchema.segmentTimestamp(key);
+        // check if timestamp is expired
+        if (timestampFromKey < observedStreamTime - retentionPeriod + 1)
+            return null;
```

Comment:
We might be able to print a debug log here to mention the value is expired so won't return. Same as others.

Reply:
Any thoughts for the comment?

Reply:
Added debug statement here.